1. ¿Cuál es la salida del siguiente fragmento de código?


In [ ]:
x = 1
y = 2
z = 3
z = x
x = y
y = z
print(x, y)

a) 2 2

b) 1 2

c) 1 1

d) 2 1

2. ¿Cuál es la salida del siguiente fragmento de código si se introduce un 2 y un 4 respectivamente?

In [1]:
x = input()
y = input()
print(x + y)

24


a) 6

b) 4

c) 2

d) 24

3. ¿Cuál es la salida del siguiente fragmento de código si se introduce un 3 y un 6 respectivamente?

In [2]:
x = input()
y = int(input())

print(x * y)

333333


a) 36

b) 666

c) 18

d) 333333

4. ¿Qué se imprime en pantalla al ejecutar el siguiente fragmento de código?

In [4]:
lista = [1, 2, 3, 4, 5]
print(lista[0:3])
print(lista[-1])
print(lista[1:4])
print(lista[1:4:2])
print(lista[1:])
print(lista[:3])
print(lista[::-1])
print(lista[1:-1:2])
print(lista[-2:1:-1])

[1, 2, 3]
5
[2, 3, 4]
[2, 4]
[2, 3, 4, 5]
[1, 2, 3]
[5, 4, 3, 2, 1]
[2, 4]
[4, 3]


5. ¿Qué se imprime por pantalla al ejecutar el siguiente fragmento de código?

In [5]:
nums = [1, 2, 3]
vals = nums
del vals[1:2]
print(nums, vals)

[1, 3] [1, 3]


In [6]:
# Si quiero hacer una copia
nums = [1, 2, 3]
vals = [x for x in nums]
del vals[1:2]
print(nums, vals)

[1, 2, 3] [1, 3]


6. ¿Qué valor será asignado a la variable x al ejecutar el siguiente fragmento de código?

In [ ]:
z = 10
y = 0
x = y < z and z > y or y > z and z < y
print(x)

7. ¿Qué valores contendrá la lista l al ejecutar el siguiente fragmento de código?

In [7]:
l = [10 * x for x in range(5)]

print(l)

[0, 10, 20, 30, 40]


8. Convierte este bucle for en un bucle while.

In [ ]:
for i in range(5):
    print(i)

In [ ]:
i = 0
while i < 5:
    print(i)
    i += 1

9. ¿Qué se imprime por pantalla al ejecutar el siguiente fragmento de código?

```	
x = 1
if x = 1:
    print("1")
```

In [8]:
x = 1
if x = 1:
    print("1")

SyntaxError: invalid syntax. Maybe you meant '==' or ':=' instead of '='? (4266369148.py, line 2)

10. ¿Qué se imprime por pantalla al ejecutar el siguiente fragmento de código?

In [9]:
for i in range(1, 5):
    for j in range(i):
        print(i, end='')
    print()

1
22
333
4444


11. ¿Cuántos * enviará el siguiente fragmento de código a la consola?

In [ ]:
i = 0
while i <= 10:
    if i % 2 == 0:
        break
    i += 1
    print("*")

12. ¿Cuál es el resultado de ejecutar el siguiente fragmento de código?

In [10]:
i = 0
while i <= 10:
    if i % 2 == 0:
        continue
    i += 1
    print("*")

*
*
*
*
*
*


In [ ]:
i = 0
while i <= 10:
    i += 1
    if i % 2 == 0:
        continue
    print("*")

13. ¿Qué valores tienen x, y, z tras ejecutar el siguiente fragmento de código?

In [11]:
x = 1
y = 2
z = 3
x, y, z = y, z, x
print(x, y, z)

2 3 1


14. Escribe este fragmento de código en una sola línea.

In [ ]:
lista = []
for i in range(1, 4):
    lista.append(i*i)

In [ ]:
lista = [i*i for i in range(1,4)]

15. ¿Qué valor devuelven las siguientes llamadas a la función?

In [12]:
def funcion(x, y = 2):
    return x*y

print(funcion(3))
print(funcion(3, 3))

6
9


16. ¿Que imprime el siguiente fragmento de código?

In [13]:
def rectangulo(base, altura):
    area = base * altura
    perimetro = 2 * base + 2 * altura
    return area, perimetro

print(rectangulo(2, 3))

(6, 10)


17. Una función definida de esta manera:

In [ ]:
def funcion(x = 0):
    return 2 * x

a) Puede ser invocada sin argumentos.

b) Puede ser invocada con un argumento.

c) a y b son correctas.

d) Ninguna de las anteriores son correctas.

18. ¿Cuál es la salida del siguiente fragmento de código?

In [14]:
def funcion(x):
    if x % 2 == 0:
        return True
    
print(funcion(3))

None


19. Para el siguiente fragmento de código, si el usuario introduce 0, ¿qué se imprime por pantalla? ¿Y si introduce 1.5?

In [17]:
try:
    x = int(input("Escribe un numero: "))
    resultado = 10 / x
    print("El resultado es: ", resultado)
except ZeroDivisionError:
    print("Division entre cero")
except ValueError:
    print("Debes escribir un numero")
except:
    print("Error")

Debes escribir un numero


20. Escribe un bucle que recorra la lista l mostrando por pantalla cada nombre de ciudad y la posición en la que se encuentra en la lista.

```python
Madrid 0
Valencia 1
Barcelona 2
Sevilla 3
```

In [18]:
ciudades = ["Madrid", "Valencia", "Barcelona", "Sevilla"]

# Aquí va tu código
for i in range(len(ciudades)):
    print(ciudades[i], i)


Madrid 0
Valencia 1
Barcelona 2
Sevilla 3


21. ¿Qué se imprime por pantalla al ejecutar el siguiente fragmento de código?

In [20]:
cadena = "Hola Mundo"
cadena.replace("Mundo", "Python")
print(cadena)

Hola Python


22. ¿Qué se imprime por pantalla al ejecutar el siguiente fragmento de código?

```python
ciudades = ("Madrid", "Valencia", "Barcelona", "Sevilla")
del ciudades[0]
print(ciudades)
```

In [ ]:
ciudades = ("Madrid", "Valencia", "Barcelona", "Sevilla")
del ciudades[0]
print(ciudades)